be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -86,45 +136,62 @@ class SinkBuffer {
         }
     }
 
-    bool is_full() const { return _pending_chunks.get_size() >= config::pipeline_io_cache_size; }
+    bool is_full() const {
+        // TODO(hcf) if one channel is congested, it may cause all other channel unwritable
+        // std::queue' read is concurrent safe without mutex
+        for (auto& cache : _caches) {
+            if (cache.size() > config::pipeline_io_cache_size) {
+                return true;
+            }
+        }
+        return false;
+    }
 
-    bool is_finished() const { return (_pending_chunks.empty() && !_closure->has_in_flight_rpc()) || _is_cancelled; }
+    bool is_finished() const { return _in_flight_rpc_num == 0 || _is_cancelled; }
```

Comment:
should check the buffer request is empty

Reply:
ok,fixed